### ISSUE TITLE
Incompatible configuration

### ISSUE THREAD

**Issue by 0rex** [gnustavo/Git-Hooks#4]
Hi Gustavo and thank you for your work! While trying to implement CheckJira function i've encountered the following error: 

```
Git::More: you have incompatible configuration sections: 'checkjira' and 'githooks.checkjira'.
remote: Please, rename all variables from section 'checkjira' to the subsection 'githooks.checkjira'.
```

The whole plugin seem to be working on my machine, because without CheckJira.jiraurl/user/pass configuration, it checks comment for the issue ID (preventing from not typing it) and then just dies in the part, where credentials are needed. This error comes when i trying to configure [githooks "CheckJira"] subsection (i don't have a checkjira section in my config).

git config: 

```
[core]
        repositoryformatversion = 0
        filemode = true
        bare = true
[githooks]
        plugin = CheckJira
[githooks "CheckJira"]
        jiraurl = <url>
        jirauser = <username>
        jirapass = <password>
```

git config -l

```
core.repositoryformatversion=0
core.filemode=true
core.bare=true
githooks.plugin=CheckJira
githooks.CheckJira.jiraurl=<url>
githooks.CheckJira.jirauser=<username>
githooks.CheckJira.jirapass=<password>
```

no global config is present.

perl -v

```
This is perl, v5.10.1 (*) built for x86_64-linux-thread-multi
```

uname -a (it's CentOS 6.4)

```
Linux 2.6.32-358.el6.x86_64 #1 SMP Fri Feb 22 00:31:26 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux
```

Can you assist me on this matter? What am i doing wrong? Thank you in advance!

**Comment by gnustavo** [comment]
Hi 0rex. I can't see anything wrong with your configuration. What version of Git::Hooks are you using?

**Comment by 0rex** [comment]
Thank you for the fast reply. My version of Git::Hooks is 0.046 (was freshly installed). I had to use force key while installing it, because one of the dependencies required external connection to complete tests, and my lab was behind a firewall (i am not yet ready to test this script on my production environment), but i recall that other non-network related tests were passed. I just can't figure out from where it can take a 'checkjira' section. Is there a way how can i provide you with more debugging info? Today i'll try it on other machine with direct internet access to see if it'll work.

**Comment by gnustavo** [comment]
I found the problem! Please, make this change to Git/More.pm and see if it works for you. If it does I'll make a new release.

```
diff --git a/lib/Git/More.pm b/lib/Git/More.pm
index 8c0eba8..ae1dfa0 100644
--- a/lib/Git/More.pm
+++ b/lib/Git/More.pm
@@ -66,7 +66,7 @@ sub _compatibilize_config {

     foreach my $var (qw/admin userenv/) {
         next if exists $config->{githooks}{$var};
-        foreach my $plugin (qw/checkacls checkjira/) {
+        foreach my $plugin (grep {exists $config->{$_}} qw/checkacls checkjira/) {
             if (exists $config->{$plugin}{$var}) {
                 $config->{githooks}{$var} = $config->{$plugin}{$var};
                 next;
```

**Comment by 0rex** [comment]
Thank you very much for your work! This fix works like a charm! I'm looking forward to see the new release soon.

**Comment by gnustavo** [comment]
Thank you for reporting the problem. I'll try to make the new release today.

**Comment by gnustavo** [comment]
Just to let you know that I released Git::Hooks 0.047. It should appear shortly at CPAN.
